### DISCUSSION TITLE
How to move directional light's projection plane.

### DISCUSSION THREAD

**Discussion by N8python** [enable3d/enable3d#179] [Help / Questions] [answered]
I was playing around with the default lighting in `warpSpeed`, and I'm trying to figure out how to move the shadow map of a directional light - I tried something along the lines of this:
```js
this.directionalLight.position.set(this.player.position.x + 25, this.player.position.y + 50, this.player.position.z + 25);
this.directionalLight.shadow.camera.top = (this.player.position.x + 0.5);
this.directionalLight.shadow.camera.bottom = (this.player.position.x - 0.5);
this.directionalLight.shadow.camera.left = (this.player.position.z - 0.5);
this.directionalLight.shadow.camera.right = (this.player.position.z + 0.5);
this.directionalLight.shadow.camera.updateProjectionMatrix();
```
However, this doesn't properly move the shadow map and the projection becomes weird - it fails to stick with the player, and often moves in a strange nonlinear manner. Is there other math I need - or another attribute I should change?

Is there something that I'm not properly changing or altering?

**Comment by yandeu** [comment, marked as answer]
See https://enable3d.io/examples/raycast-vehicle.html

**Reply by N8python** [reply]
Realized - what I was doing wrong - I needed to set the `target` of the light - thanks for the demo anyway!

**Reply by yandeu** [reply]
You're welcome :)
